### DISCUSSION TITLE
Does the forces.f90 file require modifications for 3D cases ?

### DISCUSSION THREAD

**Discussion by gauravity30** [xcompact3d/Incompact3d#256] [Q&A] [answered]
Hello,
I am trying to validate the flow around a sphere at Re=3700. I have used a coarse domain of 5D x 3D x 3D with discretization (321, 321, 120) and slip boundary conditions in y-directions whereas periodic b.c. in z-direction (I have attached the input.i3d file for your reference). [inputi3d.txt](https://github.com/xcompact3d/Incompact3d/files/14760963/inputi3d.txt)

The issue is the values of CL and CD, I am getting doesn't feel right especially if I use stretching in y-direction (iibm=2 for stretching case) while keeping everything else same.  Does the `force.f90` file require any modifications to calculate the CL and CD for 3D cases ? From what I understood, the subroutine is calculating the CL and CD for each plane in z direction and averaging it.

Without Stretching: 
![image](https://github.com/xcompact3d/Incompact3d/assets/53783610/5ae0dc3e-caff-4f7d-8549-db04f3ccfc01)

With Stretching: 
![image](https://github.com/xcompact3d/Incompact3d/assets/53783610/77942ac5-c84d-486a-b72b-fab4bb2347dc)

Thank you for the help and suggestions !

**Comment by slaizet** [comment, marked as answer]
Hi,
The current implementation in the master branch is designed for extruded shapes in the Z-direction (like the cylinder), so you would need to change the code to accommodate for the volume control surfaces in the Z-direction.
Cheers,
Sylvain

**Reply by gauravity30** [reply]
Hello ! 
Thank you for the response. Can you please share the source for the method of computation of the forces ? I am not able to find the reference mentioned in the forces.f90 file.

**Comment by slaizet** [comment]
Have a look at this report for the equations but you just need to do in Z-direction what is done in the X- and Y-direction.
Best,
Sylvain
[HOESSLER.pdf](https://github.com/xcompact3d/Incompact3d/files/14775212/HOESSLER.pdf)

**Reply by gauravity30** [reply]
Thank you for the reference ! I will have a look at it.

**Reply by gauravity30** [reply]
Hello @slaizet !
I tried modified the `force.f90` file based on the equations provided in the reference. This is the result I got from computing the lift and drag using the already available code (2D CV) and using the new code I added (3D CV). The simulation is for a flow around cylinder. The input file used is '[input_test.i3d](https://github.com/xcompact3d/Incompact3d/blob/11fda8226929b84950320c11ff94a1540fd9fb9a/examples/Cylinder/input_test.i3d)'.
![image](https://github.com/xcompact3d/Incompact3d/assets/53783610/e48c1cc4-25d5-48b0-8451-ec8d7b35e744)

Thanks a lot ! I wish to contribute to xcompact3d by completing the code and making it available for others users too. So, I would need some guidance from your side. 

Edit: Earlier, I had faced some issues related to parallelization which I solved. Now, the code works fine on both my local PC and HPC.
